lib/stacks/lib/common-stack.ts

```diff
@@ -0,0 +1,62 @@
+import * as cdk from "aws-cdk-lib";
+import * as cdkConstructs from "constructs";
+import * as iam from 'aws-cdk-lib/aws-iam';
+import * as nag from "cdk-nag";
+
+export interface StacksCommonStackProps extends cdk.StackProps {
+
+}
+
+export class StacksCommonStack extends cdk.Stack {
+    AWS_STACKNAME = cdk.Stack.of(this).stackName;
+    AWS_ACCOUNT_ID = cdk.Stack.of(this).account;
+
+    constructor(scope: cdkConstructs.Construct, id: string, props: StacksCommonStackProps) {
+        super(scope, id, props);
+
+        const region = cdk.Stack.of(this).region;
+
+        const instanceRole = new iam.Role(this, `node-role`, {
+            assumedBy: new iam.ServicePrincipal("ec2.amazonaws.com"),
+            managedPolicies: [
+                iam.ManagedPolicy.fromAwsManagedPolicyName("SecretsManagerReadWrite"),
+                iam.ManagedPolicy.fromAwsManagedPolicyName("AmazonSSMManagedInstanceCore"),
+                iam.ManagedPolicy.fromAwsManagedPolicyName("CloudWatchAgentServerPolicy"),
+            ],
+        });
```

Comment:
Also @frbrkoala why does the instance role need to be deployed with a different stack? Are there persistent resources from the other stacks that need this role to maintain permissions on them? I can't think of a justification so I bet there's something for me to learn here.

Reply:
That's because both single and ha stacks can be deployed separately but re-use the same role so that there will be a single place to control it. Additionally, I saw sometimes customers prefer to use separate stack for roles because of separation of duties in their teams (people managing security are different from people managing apps). Also that initial stack is used for other chains to deploy other common resources such as S3 buckets to store snapshots.